dask/dataframe/tests/test_accessors.py

```diff
@@ -156,14 +156,15 @@ def test_str_accessor(df_ddf):
     # with `boolean` dtype, while using object strings returns a `bool`. We cast
     # the pandas DataFrame here to ensure pandas and Dask return the same dtype.
     ctx = contextlib.nullcontext()
-    if pyarrow_strings_enabled():
+    if pyarrow_strings_enabled() and not PANDAS_GE_210:
         df.str_col = to_pyarrow_string(df.str_col)
         ctx = pytest.warns(
             pd.errors.PerformanceWarning, match="Falling back on a non-pyarrow"
         )
     assert_eq(
         ddf.str_col.str.contains("a"),
         df.str_col.str.contains("a"),
+        check_dtype=False,
```

Comment:
Our result is correct, so is the pandas result, but since we infer strings now, we get something different. That's expected